### DISCUSSION TITLE
Hey everyone, I was wondering if it's possible to validate a property before it's being updated, and if the validation fails the property should not be updated.

### DISCUSSION THREAD

**Discussion by hacktolove** [livewire/livewire#7363] [1. Help] [answered]
Hey everyone, I was wondering if it's possible to validate a property before it's being updated, and if the validation fails the property should not be updated. 
I'm currently trying to accomplish this with
``` 
public function updating(string $attribute): void
    {
        $this->validateOnly($attribute);
    }
```

But even if validation fails, the property stills gets updated. This is not what I want, as demonstrated in this example below, it will give errors when `age` is set to a string by an input field, because the typed property is `int`. 

```
class Cat extends Component {

    public int $age = 0;

    public function rules(): array
    {
        return [
            'age' => ['integer']
        ];
    }

    public function updating(string $attribute): void
    {
        $this->validateOnly($attribute);
    }
}
```

Does anyone know if there's a solution for this?

_Originally posted by @DanielGSoftware in https://github.com/livewire/livewire/discussions/2349_

**Comment by hacktolove** [comment]
Having this issue in Livewire v3

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/livewire/livewire/discussions/3147

**Comment by mintalicious** [comment, marked as answer]
The validation will always be okay, because in updating() the component property has not been updated yet, but $this->validateOnly() tries to validate the currently set value of the property. So, what the code does is validating an old state.
On the other hand $age is declared as int. Assigning non-int values to it won't work.

You could use a custom validator with the subset of component rules corresponding to the $field being updated:
```
public function updating($field, $value) {
    validator(
        [$field => $value],
        [$field => $this->rules()[$field] ?? []]
    )->validate();
}
```

**Reply by hacktolove** [reply]
That did the trick , thanks for your help @mintalicious .
